File: 20221102/모범답안/20221102_05/MatrixFullWith_test.js
import assert from 'assert'
import solution from './MatrixFullWith.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const matrix = [[true, true, true, true], [true, true, false, true], [true, false, true, true], [true, true, true, true]]
    const b = true

    // when
    const result = solution(matrix, b)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const matrix = [[true, true], [true, true]]
    const b = true

    // when
    const result = solution(matrix, b)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const matrix = [[true]]
    const b = false

    // when
    const result = solution(matrix, b)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const matrix = [[false]]
    const b = true

    // when
    const result = solution(matrix, b)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const matrix = [[true, true], [true, true]]
    const b = false

    // when
    const result = solution(matrix, b)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const matrix = [[false, true, false], [true, false, true], [false, true, false]]
    const b = true

    // when
    const result = solution(matrix)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });
})

File: 20221102/모범답안/20221102_05/MatrixFullWith_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.easy.MatrixFullWith.MatrixFullWith import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        matrix = [[True, True, True, True], [True, True, False, True], [True, False, True, True], [True, True, True, True]]
        b = True

        # when
        result = solution(matrix, b)

        # then
        self.assertEqual(False, result)

    def test_2(self):
        # given
        matrix = [[True, True], [True, True]]
        b = True

        # when
        result = solution(matrix, b)

        # then
        self.assertEqual(True, result)

    def test_3(self):
        # given
        matrix = [[True]]
        b = False

        # when
        result = solution(matrix, b)

        # then
        self.assertEqual(False, result)

    def test_4(self):
        # given
        matrix = [[False]]
        b = True

        # when
        result = solution(matrix, b)

        # then
        self.assertEqual(False, result)

    def test_5(self):
        # given
        matrix = [[True, True], [True, True]]
        b = False

        # when
        result = solution(matrix, b)

        # then
        self.assertEqual(False, result)

    def test_6(self):
        # given
        matrix = [[False, True, False], [True, False, True], [False, True, False]]
        b = True

        # when
        result = solution(matrix, b)

        # then
        self.assertEqual(False, result)


if __name__ == '__main__':
    unittest.main()
